tests/test_plugin.py

```diff
@@ -777,7 +777,7 @@ def test_channel_state_changed_bilateral(node_factory, bitcoind):
 
     The misc_notifications.py plugin logs `channel_state_changed` events.
     """
-    opts = {"plugin": os.path.join(os.getcwd(), "tests/plugins/misc_notifications.py")}
+    opts = {"plugin": os.path.join(os.getcwd(), "tests/plugins/misc_notifications.py"), "disable-plugin": "cln-grpc"}
```

Comment:
Why do we need to disable cln-grpc for tests `test_channel_state_changed_unilateral` & `test_channel_state_changed_bilateral`? Notifications should not be fail with cln-grpc.

Reply:
Cant really tell why, but they failthese are some of the errors I see with grpc enabledsee the following errors in the logs if I run them with grpc enabled:
```
lightningd-2 2024-10-07T19:39:06.394Z DEBUG   plugin-cln-grpc: Received a message: CustomNotification(Object {\"jsonrpc\": String(\"2.0\"), \"method\": String(\"block_added\"), \"params\": Object {\"block_added\": Object {\"hash\": String(\"004cb0c92dfb586bd55a3819e6ecf5e578d9ddbb09d4da65231ae4aa3bcc5f9b\"), \"height\": Number(103)}}})
lightningd-2 2024-10-07T19:39:06.394Z DEBUG   plugin-cln-grpc: Dispatching custom notification Object {\"jsonrpc\": String(\"2.0\"), \"method\": String(\"block_added\"), \"params\": Object {\"block_added\": Object {\"hash\": String(\"004cb0c92dfb586bd55a3819e6ecf5e578d9ddbb09d4da65231ae4aa3bcc5f9b\"), \"height\": Number(103)}}}
lightningd-2 2024-10-07T19:39:06.394Z UNUSUAL plugin-cln-grpc: Failed to broadcast notification SendError(BlockAdded(BlockAddedNotification { hash: 0x004cb0c92dfb586bd55a3819e6ecf5e578d9ddbb09d4da65231ae4aa3bcc5f9b, height: 103 }))
lightningd-2 2024-10-07T19:39:06.437Z DEBUG   plugin-cln-grpc: Failed to parse notification from lightningd Error(\"unknown variant `channel_state_changed`, expected one of `block_added`, `channel_open_failed`, `channel_opened`, `connect`, `custommsg`\", line: 0, column: 0)
```
```
lightningd-1 2024-10-07T19:39:06.403Z DEBUG   plugin-cln-grpc: Received a message: CustomNotification(Object {\"jsonrpc\": String(\"2.0\"), \"method\": String(\"block_added\"), \"params\": Object {\"block_added\": Object {\"hash\": String(\"004cb0c92dfb586bd55a3819e6ecf5e578d9ddbb09d4da65231ae4aa3bcc5f9b\"), \"height\": Number(103)}}})
lightningd-1 2024-10-07T19:39:06.403Z DEBUG   plugin-cln-grpc: Dispatching custom notification Object {\"jsonrpc\": String(\"2.0\"), \"method\": String(\"block_added\"), \"params\": Object {\"block_added\": Object {\"hash\": String(\"004cb0c92dfb586bd55a3819e6ecf5e578d9ddbb09d4da65231ae4aa3bcc5f9b\"), \"height\": Number(103)}}}
lightningd-1 2024-10-07T19:39:06.403Z UNUSUAL plugin-cln-grpc: Failed to broadcast notification SendError(BlockAdded(BlockAddedNotification { hash: 0x004cb0c92dfb586bd55a3819e6ecf5e578d9ddbb09d4da65231ae4aa3bcc5f9b, height: 103 }))
```
```
lightningd-2 2024-10-07T19:39:06.436Z DEBUG   plugin-cln-grpc: Received a message: CustomNotification(Object {\"jsonrpc\": String(\"2.0\"), \"method\": String(\"channel_state_changed\"), \"params\": Object {\"channel_state_changed\": Object {\"cause\": String(\"remote\"), \"channel_id\": String(\"252d1b0a1e57895e84137f28cf19ab2c35847e284c112fefdecc7afeaa5c1de7\"), \"message\": String(\"Lockin complete\"), \"new_state\": String(\"CHANNELD_NORMAL\"), \"old_state\": String(\"DUALOPEND_AWAITING_LOCKIN\"), \"peer_id\": String(\"0266e4598d1d3c415f572a8488830b60f7e744ed9235eb0b1ba93283b315c03518\"), \"short_channel_id\": String(\"103x1x1\"), \"timestamp\": String(\"2024-10-07T19:39:06.424Z\")}}})
lightningd-2 2024-10-07T19:39:06.436Z DEBUG   plugin-cln-grpc: Dispatching custom notification Object {\"jsonrpc\": String(\"2.0\"), \"method\": String(\"channel_state_changed\"), \"params\": Object {\"channel_state_changed\": Object {\"cause\": String(\"remote\"), \"channel_id\": String(\"252d1b0a1e57895e84137f28cf19ab2c35847e284c112fefdecc7afeaa5c1de7\"), \"message\": String(\"Lockin complete\"), \"new_state\": String(\"CHANNELD_NORMAL\"), \"old_state\": String(\"DUALOPEND_AWAITING_LOCKIN\"), \"peer_id\": String(\"0266e4598d1d3c415f572a8488830b60f7e744ed9235eb0b1ba93283b315c03518\"), \"short_channel_id\": String(\"103x1x1\"), \"timestamp\": String(\"2024-10-07T19:39:06.424Z\")}}}
lightningd-2 2024-10-07T19:39:06.437Z DEBUG   plugin-cln-grpc: Failed to parse notification from lightningd Error(\"unknown variant `channel_state_changed`, expected one of `block_added`, `channel_open_failed`, `channel_opened`, `connect`, `custommsg`\", line: 0, column: 0)
```

Reply:
I see. The  `wait_for_event` function simply looks for `channel_state_changed.*new_state.*` which is matched by the grpc error logs.